pkg/event/registry_test.go

```diff
@@ -0,0 +1,108 @@
+package event_test
+
+import (
+	"errors"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	k8sV1 "k8s.io/api/core/v1"
+	v1meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	submV1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	"github.com/submariner-io/submariner/pkg/event"
+	"github.com/submariner-io/submariner/pkg/event/logger"
+)
+
+const npOVNKubernetes = "OVNKubernetes"
+const npGenericKubeproxyIptables = "GenericKubeproxyIptables"
+
+var _ = Describe("Event Registry", func() {
+	When("handlers with different network plugin are added to the registry", func() {
+		var (
+			ovn, generic *TestHandler
+			registry     event.Registry
+		)
+
+		BeforeEach(func() {
+			registry = event.NewRegistry("test-registry", npGenericKubeproxyIptables)
+			ovn = NewTestHandler("handler-ovn", npOVNKubernetes)
+			generic = NewTestHandler("handler-kubeproxy", npGenericKubeproxyIptables)
+			err := registry.AddHandlers(logger.NewHandler(), ovn, generic)
+			Expect(err).NotTo(HaveOccurred())
+			cleanupTestEvents()
+		})
+
+		It("should only initialize the handlers whose network plugin matches that of the registry", func() {
+			Expect(ovn.Initialized).To(BeFalse())
+			Expect(generic.Initialized).To(BeTrue())
+		})
+
+		It("should only invoke handlers whose network plugin matches that of the registry", func() {
+			_ = registry.TransitionToGateway()
+			Expect(ovn.Events).To(HaveLen(0))
+			Expect(generic.Events).To(HaveLen(1))
+		})
+	})
+
+	When("handlers with the same network plugin are added to the registry", func() {
+		var (
+			handler1, handler2 *TestHandler
+			registry           event.Registry
+		)
+
+		BeforeEach(func() {
+			registry = event.NewRegistry("test-registry", npGenericKubeproxyIptables)
+			handler1 = NewTestHandler("handler1", npGenericKubeproxyIptables)
+			handler2 = NewTestHandler("handler2", npGenericKubeproxyIptables)
+			err := registry.AddHandlers(logger.NewHandler(), handler1, handler2)
```

Comment:
It's useful to test a "wildcarded" handler but, instead of using the logger handler, I think we should use a test handler so we can verify it.

Reply:
I'm not sure what you mean.

My intent was exercising the logging handler to make sure at least it doesnt crash.

Proper testing would mean redirecting the log to somewhere else and checking the output but I didn't think we needed to go so far.

Reply:
Irregardless of the logger handler, I meant to add a `handler3 = NewTestHandler("handler3", AnyNetworkPlugin)` and verify it also receives all the events. 

I understand the intent of adding in the logger handler here but perhaps it should be tested separately via it's own test file.

Reply:
Ahh, that's a good point!. thanks